File: Bai_16/main.cpp
#include <iostream>
#include <fstream>
#include <cmath>

int main() {
    int dem = 0;

    std::ifstream file("E:\\Language\\Baitaptonghop\\Bai_16\\DEBIEN.INP");
    int n;
    file >> n;

    for (int i = 0; i < n; ++i) {
        int l, r;
        file >> l >> r;
        int s1 = std::ceil(std::sqrt(l));
        int s2 = std::floor(std::sqrt(r));
        dem += (s2 - s1 + 1);
    }
    file.close();

    std::ofstream file_out("E:\\Language\\Baitaptonghop\\Bai_16\\DEBIEN.OUT");
    file_out << dem << "\n";
    file_out.close();

    return 0;
}

File: Bai_16/main.py
import math

dem = 0

with open('E:\Language\Baitaptonghop\Bai_16\DEBIEN.INP', 'r') as file:
    n = int(file.readline())
    for _ in range(n):
        l, r = map(int, file.readline().split())
        s1 = math.ceil(math.sqrt(l))
        s2 = math.floor(math.sqrt(r))
        dem += (s2 - s1 + 1)

with open('E:\Language\Baitaptonghop\Bai_16\DEBIEN.OUT', 'w') as file_out:
    file_out.write(str(dem) + '\n')
